src/dbsync/src/sqlite/sqlite_dbengine.cpp

```diff
@@ -944,9 +1101,61 @@ bool SQLiteDBEngine::getRowsToModify(const std::string& table,
         }
         ret = true;
     }
+    else
+    {
+        throw dbengine_error { SQL_STMT_ERROR };
+    }
+
     return ret;
 }
 
+void SQLiteDBEngine::updateSingleRow(const std::string& table,
+                                     const nlohmann::json& jsData)
+{
+    std::vector<std::string> primaryKeyList;
+    if (getPrimaryKeysFromTable(table, primaryKeyList))
+    {
+        const auto sql { buildUpdatePartialDataSqlQuery(table, jsData, primaryKeyList) };
+        const auto& stmt { getStatement(sql) };
+        const auto tableFields { m_tableFields[table] };
+
+        int32_t index { 1l };
+        for (const auto& dataValue : jsData)
+        {
+            for (auto it = dataValue.begin(); it != dataValue.end(); ++it)
+            {
+                for (const auto& field : tableFields)
```

Comment:
this can be a lambda defined in the beginning of the method that accepts a boolean to select whether to bind PK or not.
it could be express using std::for_each()

Reply:
Will do.